lib/util/location.ex

```diff
@@ -97,6 +99,112 @@ defmodule Util.Location do
+
+  ## Examples
+      iex> Util.Location.displacement_from(
+      ...>    Util.Location.new(42, -71.0001),
+      ...>    Util.Location.new(42, -71.0002)
+      ...> )
+      %Util.Vector2d{x: 8.263404849676181, y: 0.0}
+
+      iex> Util.Location.displacement_from(
+      ...>    Util.Location.new(42.0001, -71.0002),
+      ...>    Util.Location.new(42, -71.0002)
+      ...> )
+      %Util.Vector2d{x: 0.0, y: 11.119508023696506}
+
+  If the displacement is south and/or west, then the `x` or `y` coordinate will be negative
+
+  ## Example
+      iex> Util.Location.displacement_from(
+      ...>    Util.Location.new(42.0001, -71.0002),
+      ...>    Util.Location.new(42.0002, -71.0001)
+      ...> )
+      %Util.Vector2d{x: -8.26337887771932, y: -11.119508022906418}
+  """
+  @spec displacement_from(to :: __MODULE__.From.t(), from :: __MODULE__.From.t()) :: Vector2d.t()
+  def displacement_from(
+        %__MODULE__{latitude: to_latitude, longitude: to_longitude},
+        %__MODULE__{latitude: from_latitude, longitude: from_longitude}
+      ) do
+    latitude_scale_factor =
+      1000 *
+        distance(
+          new(from_latitude, from_longitude),
+          new(from_latitude + 0.001, from_longitude)
+        )
+
+    longitude_scale_factor =
+      1000 *
+        distance(
+          new(from_latitude, from_longitude),
+          new(from_latitude, from_longitude + 0.001)
+        )
+
+    longitude_diff = to_longitude - from_longitude
+    latitude_diff = to_latitude - from_latitude
+
+    %Vector2d{
+      x: longitude_diff * longitude_scale_factor,
+      y: latitude_diff * latitude_scale_factor
+    }
+  end
+
+  @doc """
+  Returns the coordinates you get by displacing the first argument (a lat/long coordinate pair)
+  `x` meters east and `y` meters north.
+
+  ## Examples
+      iex> Util.Location.displace_by(
+      ...>    Util.Location.new(42.0, -71.0),
+      ...>    %Util.Vector2d{x: 10.0, y: 0.0}
+      ...> )
+      %Util.Location{latitude: 42.0, longitude: -70.99987898450841}
+
+      iex> Util.Location.displace_by(
+      ...>    Util.Location.new(42.0, -71.0),
+      ...>    %Util.Vector2d{x: 0.0, y: 10.0}
+      ...> )
+      %Util.Location{latitude: 42.000089932036374, longitude: -71.0}
+
+  Negative values of `x` and `y` correspond to west and south displacements.
+
+  ## Example
+      iex> Util.Location.displace_by(
+      ...>    Util.Location.new(42.0, -71.0),
+      ...>    %Util.Vector2d{x: -10.0, y: -10.0}
+      ...> )
+      %Util.Location{latitude: 41.999910067963626, longitude: -71.00012101549159}
+  """
+  @spec displace_by(loc :: __MODULE__.t(), displacement :: Util.Vector2d.t()) :: __MODULE__.t()
+  def displace_by(
+        %__MODULE__{latitude: from_latitude, longitude: from_longitude},
+        %Vector2d{x: x, y: y}
+      ) do
+    latitude_scale_factor =
+      1000 *
+        distance(
+          new(from_latitude, from_longitude),
+          new(from_latitude + 0.001, from_longitude)
+        )
```

Comment:
It seems that the choice of `1000` versus another multiplier might be a bit arbitrary, except that this function needs to be the mathy inverse of the `displacement_from` function. Like, `displacement_from` gets a vector by calculating a scale_factor and multiplying the lat-lng by that factor, and this function undoes that scaling by dividing the lat-lng by the same scale factor. I think this is readable and comprehensible enough, but I can also see these two functions being refactored so that the scale_factor is calculated in a separate function that both `displacement_from` and `displace_by` use. Up to you if you like that or not!

Reply:
Oh yeah good call. That had bothered me a bit, but then I completely forgot.

Reply:
I believe that [this](https://github.com/mbta/skate/pull/2510/commits/9d07fd4bcdee050de2ea6720a15d1e42e9e98e94) should address that. I went with `small_increment` to represent `0.001` (and divided by `small_increment` instead of multiplying by `1000`) to make it clearer that it's just a random small number, and also because now you don't have to guess at the relationship between `1000` and `0.001`.